clang/runtime/dpct-rt/include/dpl_extras/dpcpp_extensions.h.inc

```diff
@@ -677,20 +677,20 @@ template <typename... _Args> constexpr auto __joint_reduce(_Args... __args) {
 /// that are one past the last element in each segment \param binary_op functor
 /// that implements the binary operation used to perform the scan. \param init
 /// initial value of the reduction for each segment.
-template <int GROUP_SIZE, typename T, class BinaryOperation>
+template <int GROUP_SIZE, typename T, typename Ptr, class BinaryOperation>
 void segmented_reduce(sycl::queue queue, T *inputs, T *outputs,
-                      size_t segment_count, uint32_t *begin_offsets,
-                      uint32_t *end_offsets, BinaryOperation binary_op,
-                      T init) {
+                      size_t segment_count, Ptr begin_offsets, Ptr end_offsets,
```

Comment:
Given the input value data is provided using T* should the offsets be provided using Offset* instead of Ptr? This would make the use of the template parameters consistent and would eliminate the need for the OffsetT type definition on line 618. The current implementation is correct, but the inconsistent use of template parameters for types that are required to be pointers makes the code a bit more difficult to follow.

Reply:
Done